### ISSUE TITLE
make' do not handle spaces in paths

### ISSUE THREAD

**Issue by GitHubIsToxic** [wxWidgets/wxWidgets#24268]
### Build System Used

I build wxWidgets and/or my application using:

- [ ] CMake
- [x] configure
- [ ] mingw32-make with makefile.gcc
- [ ] MSBuild (Microsoft Visual Studio solution file)
- [ ] nmake with makefile.vc
- [ ] Xcode

### Description

Following these exact instructions "MinGW Compilation" > "Using configure":
https://docs.wxwidgets.org/stable/plat_msw_install.html

The "$ configure" step completes without issue (I have not attached .log for privacy reasons).

Running "make" will result in many lines with the following error:
```
Makefile:14249: warning: overriding recipe for target '/home/username/.wine32/drive_c/Program'
Makefile:14177: warning: ignoring old recipe for target '/home/username/.wine32/drive_c/Program'
```
And ultimately fails on:
```
make: /home/username/.wine32/drive_c/Program: Command not found
Makefile:37244: recipe for target 'gldll_glcmn.o' failed
```

The wxWidgets library location is this:
```/home/username/.wine32/drive_c/Program Files/very long path snipped/source/external/wxWidgets-3.2.4/```

Due to this project path being inside a SVN working copy inside a Wine directory, it will always contain spaces, so this is not easy to work around by moving/renaming the directory, this will break other aspects of the project.

My current workaround is to build wxWidgets outside the project directory and then copying it back, but this is cumbersome and error-prone due to files needing to be versioned in our project.

### Platform and version information

- wxWidgets version: 3.2.4
- wxWidgets port: wxGTK?
- OS: Linux Mint 19 Cinnamon
- Compiler: gcc 7.5.0

**Comment by GitHubIsToxic** [comment]
I should note that, I'm building with gcc here, not Mingw-w64 gcc, as the Wiki documentation page omits proper instructions for Mingw-w64 (but that's another issue).

**Comment by vadz** [comment]
Sorry, I think that using spaces in paths with makefiles and other traditional tools is just asking for trouble and don't plan to spend any effort on fixing this.

Of course, if somebody else can provide a reasonably simple fix for it (and ideally some way to ensure that it remains fixed, e.g. change one of the CI builds using MinGW to use a file with spaces), I'd be glad to apply it.

In the meanwhile I can only recommend using paths without spaces or, perhaps, if they work in your configuration (I'm not sure to understand what you're doing, i.e. whether you're using Wine or not), create a symlink/junction from a path without spaces to your actual path.

**Comment by GitHubIsToxic** [comment]
No, it's just that the path to the wxWidgets source directory just contains some spaces. I was assuming that a regular wxWidgets 'build' with default configuration was supposed to handle spaces just fine. Especially because 'configure' completes without issue or warning. Perhaps if spaces are not decided to be not allowed, it would be nice to abort with a warning, rather than throw a long list of ambiguous errors.

It could just be a case of double quotes missing on a few lines in the makefile, maybe once I've got everything compiling reliably I'll look into it.

Thanks!

**Comment by oneeyeman1** [comment]
Using spaces in the path that have wxWidgets for building was never officially supported.

I think there was even a note somewhere in the docs warning about that.

However I don't understand why the project will always contain spaces - cloning everything is under you control. You can in fact clone/copy wxWidgets sources anywhere on the system, build it and then use proper path to point to your build directory.

That's not to say that spaces in the path can't be handled.

Thank you.

**Comment by GitHubIsToxic** [comment]
The developed applications in our project (some parts out of our control) are required to run, tested and debugged inside the Windows Program File directory, a directory path which always contains spaces. Some of our users run it natively on Windows, some on Wine. The application I'm building is cross platform, with the Windows build (including wxWidgets and other libraries) compiled using Mingw-w64. The Linux build is using precompiled shared libraries provided by the OS, so I had not run into this problem yet.

One problem I'm now also running into is, even once wxWidgets is compiled (via symlink workaround mentioned above), wx-config still fails if the temporary symlink is removed after compilation:

```  *** Error: Directory '/media/username/sdk/wxWidgets-3.2.4/build-msw/..'
             no longer exists.
```
So there's the issue of absolute paths being stored in resulting configurations as well. This would be avoided if the workaround wasn't needed in the first place. I don't see why all paths couldn't be relative to the wxWidgets source folder, it would make things much more 'portable', and avoid the spaces in path problem as well.

**Comment by oneeyeman1** [comment]
@GitHubIsToxic ,
I understand that you should rund that from "Program Files ". But you don't have to compile it there.
What you can do is grab the sources, unpack it into c:\wxWidgets,  build the library, then use -I and -L options to set the search path as appropriate in order to compile your binaries.
When it is time to run it you copy wx libraries to whenever you require them to be.

All the above does not mean spaces in the path shouldn't be handled...

Thank you.

**Comment by GitHubIsToxic** [comment]
Yes, of course it's always possible to work around, but issues with spaces-in-paths is kind of a early 90's computing issue that you'd expect to have been overcome by now.
Anyway, besides the inconvenience, other libraries (so far) and our own code don't have similar problems, wxWidgets stands out in this.

Thanks for your time.

**Comment by GitHubIsToxic** [comment]
Also, by manually editing Makefile generated by "configure", changing wx_top_builddir to:
```wx_top_builddir = "./"```
Fixes compilation, although it does fail at the end, it gives me the .a files I need for my cross-compiled application to link properly. So it should not take much work to support spaces in paths, by simply using only relative paths. That's even better and more robust than handling spaces IMO.
